src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectReloadManager.cs

```diff
@@ -0,0 +1,508 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.ProjectSystem.VS.UI;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    // <summary>
+    // ProjectReloadManager
+    //
+    // VS wide CPS component which manages project reloads on behalf of projects. As projects are loaded they register with this component
+    // which will add that project to its list of watched projects. When the project file changes outside of VS, this object will call into the project to
+    // perform a silent reload. Using a central object allows batching of reload messages, especially when silent reloads fail due to problems with
+    // the project file, or the file in memory is dirty requiring user intervention. It also permits handling scenarios such as automatically reloading the
+    // projects which fail the solution level reload, the next time the project file changes.
+    // </summary>
+    [Export(typeof(IProjectReloadManager))]
+    [AppliesTo("HandlesOwnReload")]
```

Comment:
This is defined in CPS but we haven't picked up that version. I'll add a comment to the code to switch to the CPS constant when it is available.